### PR TITLE
 fix #130 Empty response for NOT_MODIFIED

### LINKED ISSUE #130: Empty response for NOT_MODIFIED (304) return for static files
When the Browser requests a file and passes the cache parameters, Netty is not sending the response.

## Normal request:
image

## Cached request:
image

image

Classpath:
```


/opt/m2-repository/org/apache/shiro/shiro-web/1.2.4/shiro-web-1.2.4.jar
/opt/m2-repository/org/apache/shiro/shiro-core/1.2.4/shiro-core-1.2.4.jar
/opt/m2-repository/commons-beanutils/commons-beanutils/1.8.3/commons-beanutils-1.8.3.jar
/opt/m2-repository/org/secnod/shiro/shiro-jersey/0.1.0/shiro-jersey-0.1.0.jar
/opt/m2-repository/ch/qos/logback/logback-classic/1.0.13/logback-classic-1.0.13.jar
/opt/m2-repository/ch/qos/logback/logback-core/1.0.13/logback-core-1.0.13.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-servers/opendevice-rest-ws-server/target/classes
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-rest-api/target/classes
/opt/m2-repository/javax/ws/rs/jsr311-api/1.1.1/jsr311-api-1.1.1.jar
/opt/m2-repository/javax/inject/javax.inject/1/javax.inject-1.jar
/media/ricardo/Dados/Codidos/Java/BIBLIOTECAS/nettosphere/server/target/classes
/opt/m2-repository/org/atmosphere/atmosphere-runtime/2.4.7/atmosphere-runtime-2.4.7.jar
/opt/m2-repository/org/atmosphere/atmosphere-guice/2.4.3/atmosphere-guice-2.4.3.jar
/opt/m2-repository/com/sun/jersey/contribs/jersey-guice/1.19/jersey-guice-1.19.jar
/opt/m2-repository/com/google/inject/extensions/guice-servlet/3.0/guice-servlet-3.0.jar
/opt/m2-repository/com/google/inject/guice/3.0/guice-3.0.jar
/opt/m2-repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar
/opt/m2-repository/org/atmosphere/atmosphere-annotations/2.4.7/atmosphere-annotations-2.4.7.jar
/opt/m2-repository/org/atmosphere/atmosphere-jersey/2.4.7/atmosphere-jersey-2.4.7.jar
/opt/m2-repository/com/sun/jersey/jersey-server/1.19/jersey-server-1.19.jar
/opt/m2-repository/com/sun/jersey/jersey-core/1.19/jersey-core-1.19.jar
/opt/m2-repository/com/sun/jersey/jersey-servlet/1.19/jersey-servlet-1.19.jar
/opt/m2-repository/org/apache/geronimo/specs/geronimo-servlet_3.0_spec/1.0/geronimo-servlet_3.0_spec-1.0.jar
/opt/m2-repository/org/apache/oltu/oauth2/org.apache.oltu.oauth2.authzserver/1.0.1/org.apache.oltu.oauth2.authzserver-1.0.1.jar
/opt/m2-repository/org/apache/oltu/oauth2/org.apache.oltu.oauth2.common/1.0.1/org.apache.oltu.oauth2.common-1.0.1.jar
/opt/m2-repository/org/json/json/20140107/json-20140107.jar
/opt/m2-repository/org/apache/httpcomponents/httpclient/4.5.2/httpclient-4.5.2.jar
/opt/m2-repository/org/apache/httpcomponents/httpcore/4.4.4/httpcore-4.4.4.jar
/opt/m2-repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar
/opt/m2-repository/commons-codec/commons-codec/1.9/commons-codec-1.9.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-servers/opendevice-mqtt/target/classes
/opt/m2-repository/io/moquette/moquette-broker/0.10/moquette-broker-0.10.jar
/opt/m2-repository/io/netty/netty-common/4.1.12.Final/netty-common-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-buffer/4.1.12.Final/netty-buffer-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-transport/4.1.12.Final/netty-transport-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-resolver/4.1.12.Final/netty-resolver-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-handler/4.1.12.Final/netty-handler-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-codec/4.1.12.Final/netty-codec-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-codec-http/4.1.12.Final/netty-codec-http-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-codec-mqtt/4.1.12.Final/netty-codec-mqtt-4.1.12.Final.jar
/opt/m2-repository/io/netty/netty-transport-native-epoll/4.1.12.Final/netty-transport-native-epoll-4.1.12.Final-linux-x86_64.jar
/opt/m2-repository/io/netty/netty-transport-native-unix-common/4.1.12.Final/netty-transport-native-unix-common-4.1.12.Final.jar
/opt/m2-repository/io/dropwizard/metrics/metrics-core/3.2.2/metrics-core-3.2.2.jar
/opt/m2-repository/io/dropwizard/metrics/metrics-jvm/3.2.2/metrics-jvm-3.2.2.jar
/opt/m2-repository/com/librato/metrics/metrics-librato/5.1.0/metrics-librato-5.1.0.jar
/opt/m2-repository/com/librato/metrics/librato-java/2.1.0/librato-java-2.1.0.jar
/opt/m2-repository/com/bugsnag/bugsnag/3.1.6/bugsnag-3.1.6.jar
/opt/m2-repository/com/hazelcast/hazelcast/3.5.4/hazelcast-3.5.4.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-connection/opendevice-connection-stream/target/classes
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-connection/opendevice-connection-tcp/target/classes
/opt/m2-repository/br/com/criativasoft/opendevice/ext/bluecove/2.1.1-SNAPSHOT/bluecove-2.1.1-20141011.162527-1.jar
/opt/m2-repository/br/com/criativasoft/opendevice/ext/bluecove-gpl/2.1.1-SNAPSHOT/bluecove-gpl-2.1.1-20141011.162657-1.jar
/opt/m2-repository/org/scream3r/jssc/2.8.0/jssc-2.8.0.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-js-engine/target/classes
/opt/m2-repository/org/mapdb/mapdb/1.0.10-SNAPSHOT/mapdb-1.0.10-SNAPSHOT.jar
/opt/m2-repository/com/h2database/h2/1.4.192/h2-1.4.192.jar
/opt/m2-repository/org/hibernate/hibernate-entitymanager/5.0.9.Final/hibernate-entitymanager-5.0.9.Final.jar
/opt/m2-repository/org/jboss/logging/jboss-logging/3.3.0.Final/jboss-logging-3.3.0.Final.jar
/opt/m2-repository/org/hibernate/hibernate-core/5.0.9.Final/hibernate-core-5.0.9.Final.jar
/opt/m2-repository/antlr/antlr/2.7.7/antlr-2.7.7.jar
/opt/m2-repository/org/jboss/jandex/2.0.0.Final/jandex-2.0.0.Final.jar
/opt/m2-repository/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar
/opt/m2-repository/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2.jar
/opt/m2-repository/org/hibernate/common/hibernate-commons-annotations/5.0.1.Final/hibernate-commons-annotations-5.0.1.Final.jar
/opt/m2-repository/org/hibernate/javax/persistence/hibernate-jpa-2.1-api/1.0.0.Final/hibernate-jpa-2.1-api-1.0.0.Final.jar
/opt/m2-repository/org/javassist/javassist/3.18.1-GA/javassist-3.18.1-GA.jar
/opt/m2-repository/org/apache/geronimo/specs/geronimo-jta_1.1_spec/1.1.1/geronimo-jta_1.1_spec-1.1.1.jar
/opt/m2-repository/javax/transaction/jta/1.1/jta-1.1.jar
/opt/m2-repository/org/knowm/sundial/2.1.2-SNAPSHOT/sundial-2.1.2-20170503.074422-11.jar
/opt/m2-repository/org/flywaydb/flyway-core/4.2.0/flyway-core-4.2.0.jar
/opt/m2-repository/io/swagger/swagger-annotations/1.5.3/swagger-annotations-1.5.3.jar
/opt/m2-repository/org/slf4j/slf4j-api/1.6.1/slf4j-api-1.6.1.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-core/target/classes
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/opendevice-connection/opendevice-connection-api/target/classes
/opt/m2-repository/com/fasterxml/jackson/core/jackson-databind/2.9.3/jackson-databind-2.9.3.jar
/opt/m2-repository/com/fasterxml/jackson/core/jackson-annotations/2.9.0/jackson-annotations-2.9.0.jar
/opt/m2-repository/com/fasterxml/jackson/core/jackson-core/2.9.3/jackson-core-2.9.3.jar
/media/ricardo/Dados/Codidos/Java/Projetos/OpenDevice/extensions/sonoff-bridge/target/classes
/opt/m2-repository/com/diozero/diozero-core/0.11-SNAPSHOT/diozero-core-0.11-20180403.172141-17.jar
/opt/m2-repository/org/apache/commons/commons-math3/3.6.1/commons-math3-3.6.1.jar
/opt/m2-repository/com/diozero/diozero-provider-pigpio/0.11-SNAPSHOT/diozero-provider-pigpio-0.11-20180403.172312-17.jar
/opt/m2-repository/uk/pigpioj/pigpioj-java/2.2/pigpioj-java-2.2.jar
/opt/m2-repository/io/netty/netty-all/4.1.9.Final/netty-all-4.1.9.Final.jar
/opt/m2-repository/org/tinylog/tinylog/1.3.3/tinylog-1.3.3.jar
/opt/m2-repository/com/sun/jersey/contribs/jersey-multipart/1.19/jersey-multipart-1.19.jar
/opt/m2-repository/org/jvnet/mimepull/mimepull/1.9.3/mimepull-1.9.3.jar

```

### PR DESCRIPTION
**Pull request by ricardojlrufino**
2b0f1b1 fix and revert changes in  6b428a7

### CODE REVIEW

**Comment by ricardojlrufino** [comment]
@jfarcand 
this is necessary every time ?
ctx.pipeline().addBefore(BridgeRuntime.class.getName(), "encoder", new HttpResponseEncoder());

**Comment by jfarcand** [comment]
@ricardojlrufino Yes it is unless it was added by default. If you remove it, does it still work?

**Comment by ricardojlrufino** [comment]
@jfarcand not work, i need it move to upper , to fiz bug in sendNotModified

**Comment by jfarcand** [comment]
@ricardojlrufino Patch looks good, but can you add a unit test? Thanks a lot!

**Comment by jfarcand** [comment]
@ricardojlrufino Patch looks good, but can you add a unit test? Thanks a lot!

**Comment by ricardojlrufino** [comment]
Sorry but I have no idea how to implement for this specific use case

**Comment by jfarcand** [comment]
@ricardojlrufino How do you test it?

**Comment by ricardojlrufino** [comment]
I am implementing the protocol of this equipment: SonOff
image

**Comment by ricardojlrufino** [comment]
(THE CACHE PROBLEM)
I'm testing directly in chrome, I hope it does the cache .... and then it starts to appear these errors of EMPTY_RESPONSE

File: server/src/main/java/org/atmosphere/nettosphere/HttpStaticFileServerHandler.java [hunk 1]
```diff
@@ -74,9 +74,7 @@
 import java.util.TimeZone;
 import java.util.regex.Pattern;
 
-import static io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
-import static io.netty.handler.codec.http.HttpHeaders.Names.IF_MODIFIED_SINCE;
-import static io.netty.handler.codec.http.HttpHeaders.Names.LOCATION;
+import static io.netty.handler.codec.http.HttpHeaders.Names.*;
 import static io.netty.handler.codec.http.HttpMethod.GET;
```
Comment (jfarcand) [inline server/src/main/java/org/atmosphere/nettosphere/HttpStaticFileServerHandler.java]: Please do not change the `import`

**Comment by jfarcand** [comment]
@ricardojlrufino OK fair enough. Can you just fix the comment and I will integrate your fix and cut a release ASAP. Thanks!

**Pull request merged**